packages/shopping/src/application.ts

```diff
@@ -101,6 +102,11 @@ export class ShoppingApplication extends BootMixin(
 
     this.bind(UserServiceBindings.USER_SERVICE).toClass(UserManagementService);
     this.add(createBindingFromClass(SecuritySpecEnhancer));
+
+    // Use JWT secret from JWTSECRET environment variable if set
+    // otherwise create a random 256 bit long secret key
+    const secret = process.env.JWTSECRET || crypto.randomBytes(32).toString();
```

Comment:
@johanschvn should probably specify the encoding of the string either `toString("hex")` or `toString("base64")`

Reply:
It is binary data - should be toString('binary'); I have changed. Using 'binary' instead of 'latin1' as I think it is easier to understand.

Reply:
Actually, now I am not sure anymore what kind of string the library used (npm package jsonwebtoken) expect. I have only used keys with printable ASCII characters.

Reply:
My best guess now is UTF8; maybe we should use 64 hex digits as the secret. Also I think I should remove "key" and only talk about secret, so not to confuse with public/private key pairs. What do you think?

Reply:
The strength of the secret only depends on the size in bits but the encoding could potentially be necessary to avoid issues when passing the secret to another library. I usually use `base64` or `hex` encoding to prevent avoid those issues.

> Also I think I should remove "key" and only talk about secret, so not to confuse with public/private key pairs. What do you think?

just calling it secret sounds good to me

Reply:
Both changes are done: dropping "key" and encoding random key as "hex".